src/Database/Type/DateTimeType.php

```diff
@@ -165,7 +169,7 @@ public function marshal($value)
                 $date = new $class($value);
                 $compare = true;
             }
-            if ($compare && $date && $date->format($this->_format) !== $value) {
+            if ($compare && $date && !$this->_compare($date, $value)) {
```

Comment:
OK. Then I'll send an alternative PR to 3.next when I have time. If it becomes to accept new date formats, the master branch would not be suitable.